scripts/prepare_network.py

```diff
@@ -233,8 +243,22 @@ def set_line_nom_max(n, s_nom_max_set=np.inf, p_nom_max_set=np.inf):
             if len(m) > 0:
                 co2limit = float(m[0]) * snakemake.config['electricity']['co2base']
                 add_co2limit(n, co2limit, Nyears)
+                logger.info("Setting CO2 limit according to wildcard value.")
             else:
                 add_co2limit(n, snakemake.config['electricity']['co2limit'], Nyears)
+                logger.info("Setting CO2 limit according to config value.")
+            break
+
+    for o in opts:
+        if "CH4L" in o:
+            m = re.findall("[0-9]*\.?[0-9]+$", o)
+            if len(m) > 0:
+                limit = float(m[0]) * 1e6
+                add_gaslimit(n, limit, Nyears)
+                logger.info("Setting gas usage limit according to wildcard value.")
+            else:
+                add_gaslimit(n, snakemake.config["electricity"].get("gaslimit"), Nyears)
+                logger.info("Setting gas usage limit according to config value.")
```

Comment:
so for the wildcard value, the limit is specified in... TWh, while the config value must be given in MWh? (release_notes indicate that limit to be given in MWh in any case, if I remember correctly)

Reply:
I made a suggestion above to make this difference in unit clear.